### ISSUE TITLE
Flag pairs as belonging to 1:1, 1:n, n:1, or n:n relationship

### ISSUE THREAD

**Issue by ezwelty** [ezwelty/rgi_links#1]
Bruce Raup suggested marking each pair as belonging to a 1:1, 1:n, n:1, or n:n relationship. This could be done with a single categorical column (e.g. `relationship_type`). It could make the link table "safer" to use by explicitly mentioning the uniqueness of a match (for 1:1) or the presence of other possible matches.

**Comment by fmaussion** [comment]
Thanks! Are you able to implement this in your scripts?

**Comment by ezwelty** [comment]
@fmaussion Function `helpers.count_pair_relations` returns the number of relatives in the relationship for each side (as two columns). This could be kept as two integer columns, or reduced to the `n:m` string format, or further reduced to {`1:1`, `1:n`, `n:1`, `n:n`}. Note that there is no single way of describing n:n relationships – I went with using the total number of directly related records from each group (see function docstring). For example, if A is contained by 1, and B overlaps both 1 and 2, these pairs are all part of a 2:2 relationship because A is related to 1 which is related to A and B, the latter of which is also related to 2. If C overlapped 2 (but not 1), this pair would also be part of a 2:2 relationship because C is related to 2 which is related to C and B, the latter of which is also related to 1 (note that A, as an indirect relative, is not considered).

I also added `helpers.label_pair_clusters`, which uses an undirected graph to find clusters of directly and indirectly related pairs. In the example above, A, B, C, 1, 2 are all part of the same cluster.

See https://github.com/ezwelty/rgi_links#compute-overlaps for a usage example.
